tests/components/samsungtv/test_config_flow.py

```diff
@@ -573,13 +621,45 @@ async def test_autodetect_none(hass, remote, remotews):
             DOMAIN, context={"source": config_entries.SOURCE_USER}, data=MOCK_USER_DATA
         )
         assert result["type"] == "abort"
-        assert result["reason"] == "cannot_connect"
+        assert result["reason"] == "not_successful"
         assert remote.call_count == 1
         assert remote.call_args_list == [
             call(AUTODETECT_LEGACY),
         ]
         assert remotews.call_count == 2
         assert remotews.call_args_list == [
-            call(**AUTODETECT_WEBSOCKET_PLAIN),
             call(**AUTODETECT_WEBSOCKET_SSL),
+            call(**AUTODETECT_WEBSOCKET_PLAIN),
         ]
+
+
```

Comment:
We should have a test that the integration is setup via zeroconf, and then the user tries to go set it up manually to make sure it aborts when it sees an config entry already exists for the same unique id

Reply:
Tests need to be deeply checked, there are a lot of issue currently.

The idea was to focus on code and complete that before starting that task